88.md

```diff
@@ -0,0 +1,126 @@
+NIP-88
+======
+
+Binary event encoding
+---------------------
+
+`draft` `optional` `author:dr-orlovsky`
+
+At the present moment nostr (NIP-1) uses text serialization of the event data,
+represented in a form of a JSON object. While being simple-to-implement and 
+web-compatible, it has some significant drawbacks:
+
+* Speed: serialization from JSON consumes most of the app time, according to [1]
+* Complex handling of binary data, which have to be encoded into Base64 or some
+  other binary-to-ascii format, increasing the problem described above and 
+  leading to more traffic consumption.
+* No explicit limits for the size of the data (i.e. it is not clear what is the
+  maximal size of the message, or the maximal number of tags), leading to DoS 
+  vulnerability and interoperability. While each relay is free to choose the
+  limits to prevent DoS attacks, the limits are not yet there, opening an attack
+  surface which may down the network for some time. Also, absence of the 
+  standard will lead that some of the relays will stop propagating some events 
+  (as a part of DoS prevention), while others will support all of them, making 
+  it hard to get the predictable and consistent user experience.
+* Protocol extensibility: with the current version of the protocol it is 
+  prohibited to add new fields to the JSON format.
+
+These all problems can be solved with introducing binary encoding to the 
+messages with strict limits to the maximal size of the elements and optional TLV
+extension stream.
+
+Specification
+-------------
+
+The proposed binary encoding must follow this spec (in pseudocode):
+```haskell
+Event ::
+    -- fields matching NIP-1:
+    Id, Pubkey, Timestamp, Kind, Tags, Content,
+    -- field matching NIP-3:
+    OtsHash,
+    -- Type-length-value encoded stream:
+    TLVs,
+    -- NIP-1 signature
+    legacy Signature
+    -- new signature over binary data (except legacy sig)
+    binary Signature
+
+Id :: [Byte ^ 32]     -- indicates fixed-size byte string of exactly 32 bytes
+Pubkey :: [Byte ^ 32]
+Timestamp :: U32 -- indicates 32-bit unsigned integer
+Kind :: U64 -- indicates 64-bit unsigned integer
+OtsHash :: [Byte ^ 32]
```

Comment:
OpenTimestamps use the document id as the hash for the proof, what's inside the ots field defined in nip3 is the base64 of the proof (basically the .ots file) and is variable length.
for more info see https://github.com/RCasatta/nostr-ots

Reply:
Thank you for getting back to me I really appreciate this solution
Programming scientists etc reverse engineering is very creative and it's always a wormhole to get out of an in-out when it comes to ethical hacking lol having good day nice to hear from you

Reply:
@RCasatta good point. 

NB: Since the concept doesn't fit the current #nostr approach (which I appreciate and which is completely fine) I moved this experimental proposal into a dedicated initiative to https://github.com/renostr

I will add the amendment you suggested there.